### PR TITLE
Refactor friendly messages

### PR DESCRIPTION
**Pull request by avitova**
No idea what is the problem with the test below.
Also we should probably change the README file after the tests with json examples are done. It took me a second to realize it all had changed its path.

### CODE REVIEW

File: lib/smart_proxy_host_reports/ansible_processor.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 class AnsibleProcessor < Processor
+  require_relative "friendly_message"
```
Comment (lzap) [inline lib/smart_proxy_host_reports/ansible_processor.rb]: Please move it outside of the class.

File: lib/smart_proxy_host_reports/friendly_message.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    return msg = @result_tree["msg"] if @result["failed"]
+
+    case @task_tree["action"]
+    when "ansible.builtin.package", "package" then msg = set_package_message
+    when "ansible.builtin.template", "template" then msg = set_template_message
+    when "ansible.builtin.service", "service" then msg = set_service_message
+    when "ansible.builtin.group", "group" then msg = set_group_message
+    when "ansible.builtin.user", "user" then msg = set_user_message
+    when "ansible.builtin.cron", "cron" then msg = set_cron_message
+    when "ansible.builtin.copy", "copy" then msg = set_copy_message
```
Comment (lzap) [inline lib/smart_proxy_host_reports/friendly_message.rb]: Please rename without `set_` prefix.

File: lib/smart_proxy_host_reports/friendly_message.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    when "ansible.builtin.command", "ansible.builtin.shell", "command", "shell" then msg = @result_tree["stdout_lines"]
+    end
+  rescue StandardError => e
+    logger.debug "Unable to parse result (#{e.message}): #{@result.inspect}"
+  ensure
+    return msg
```
Comment (lzap) [inline lib/smart_proxy_host_reports/friendly_message.rb]: Do not use `ensure`, just return `msg` in both the block and the rescue block.

File: lib/smart_proxy_host_reports/friendly_message.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    @result = result
+    @result_tree = @result["result"]
+    @task_tree = @result["task"]
+    @module_args_tree = @result_tree.dig("invocation", "module_args")
+  end
+
+  def generate_message
+    msg = "N/A"
+    return msg if @task_tree.nil? || @task_tree["action"].nil?
+    return msg = @result_tree["msg"] if @result["failed"]
```
Comment (lzap) [inline lib/smart_proxy_host_reports/friendly_message.rb]: This is not standard, just return `return @result_tree["msg"]`

File: lib/smart_proxy_reports/friendly_message.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    end
+    msg
+  rescue StandardError => e
+    logger.debug "Unable to parse result (#{e.message}): #{@result.inspect}"
+    msg
+  end
+
+  def package_message
+    detail = @result_tree["results"] || @result_tree["msg"] || "No details"
+    return "Package(s) #{@module_args_tree["name"].join(",")} are #{@module_args_tree["state"]}: #{detail}"
```
Comment (ofedoren) [inline lib/smart_proxy_reports/friendly_message.rb]: The last statement in Ruby is the return value by default.
```suggestion
    "Package(s) #{@module_args_tree["name"].join(",")} are #{@module_args_tree["state"]}: #{detail}"
```

**Review by ofedoren — COMMENTED** [review]
Thanks, @avitova! Works well, although one small nitpick, please see inline.

Also, do we want to actually have the whole output for package installation? Let's imagine that we want to install more then 3 packages, the output will be pretty huge and will take all the space. Example for 3 packages:
ScreenShot-1635942490008

I guess we can skip details, since they will be stored anyway and after https://github.com/theforeman/foreman_host_reports/pull/9 is merged, users will be able to observe the whole output in there.

One more thing: do we want to be more verbose for `--check` mode? As you can see some details are missing:
ScreenShot-1635942610662

**Comment by lzap** [comment]
@ofedoren fyi this is just a refactoring PR, we aim to improve in other PRs. I suggest to merge this as pure refactoring and then @avitova could fix the mentioned problem (good catch).

@avitova please drop the `return`, improve the commit message and we can merge.

**Comment by ofedoren** [comment]
Yes, sure, I understand that my concerns are out of the scope for this PR :)

**Comment by ares** [comment]
Since we are touching this, it would be awesome to also improve the packages output that @ofedoren captured on the screenshot so that `\n` is replaced with `<br>`. Of course in a separate PR if you find that easier.

**Review by lzap — APPROVED** [review]
> Since we are touching this, it would be awesome to also improve the packages output that @ofedoren captured on the screenshot so that \n is replaced with <br>. Of course in a separate PR if you find that easier.

Yes, @avitova please tackle that in another PR.

**Pull request merged**